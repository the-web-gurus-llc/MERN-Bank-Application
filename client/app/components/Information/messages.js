/*
 * Information Messages
 *
 * This contains all the text for the Information component.
 */

import { defineMessages } from 'react-intl';

export const scope = 'app.components.Information';

export default defineMessages({
  content: {
    id: `${scope}.content`,
    defaultMessage:
      'If you take advantage of our promotion and register your account by the end of this year, you will receive 10,00 USD from us. Accounts already created will also receive this from us.',
  },
});
