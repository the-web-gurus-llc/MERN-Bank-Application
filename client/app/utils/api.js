const API_BASE_URL = 'https://api.pietrzakadrian.com/bank';
const AUTH = 'Auth';
const USERS = 'Users';
const CURRENCIES = 'Currencies';
const BILLS = 'Bills';
const TRANSACTIONS = 'Transactions';
const MESSAGES = 'Messages';
const NOTIFICATIONS = 'Notifications';

export default {
  auth: {
    login: `${API_BASE_URL}/${AUTH}/login`,
    register: `${API_BASE_URL}/${AUTH}/register`,
    logout: `${API_BASE_URL}/${AUTH}/logout`,
    forgetPassword: `${API_BASE_URL}/${AUTH}/password/forget`,
    resetPassword: `${API_BASE_URL}/${AUTH}/password/reset`,
  },
  users: (path = '') => {
    switch (path) {
      case 'checkEmail':
        return (email) => `${API_BASE_URL}/${USERS}/${email}/checkEmail`;
      default:
        return `${API_BASE_URL}/${USERS}`;
    }
  },
  currencies: `${API_BASE_URL}/${CURRENCIES}`,
  messages: `${API_BASE_URL}/${MESSAGES}`,
  notifications: `${API_BASE_URL}/${NOTIFICATIONS}`,
  bills: (path = '') => {
    switch (path) {
      case 'amountMoney':
        return `${API_BASE_URL}/${BILLS}/amountMoney`;
      case 'accountBalance':
        return `${API_BASE_URL}/${BILLS}/accountBalance`;
      case 'accountBalanceHistory':
        return `${API_BASE_URL}/${BILLS}/accountBalanceHistory`;
      case 'search':
        return (accountBillNumber) =>
          `${API_BASE_URL}/${BILLS}/${accountBillNumber}/search`;
      default:
        return `${API_BASE_URL}/${BILLS}`;
    }
  },
  transactions: (path = '') => {
    switch (path) {
      case 'create':
        return `${API_BASE_URL}/${TRANSACTIONS}/create`;
      case 'confirm':
        return `${API_BASE_URL}/${TRANSACTIONS}/confirm`;
      case 'authorizationKey':
        return (uuid) =>
          `${API_BASE_URL}/${TRANSACTIONS}/${uuid}/authorizationKey`;
      case 'confirmationFile':
        return (uuid, locale) =>
          `${API_BASE_URL}/${TRANSACTIONS}/${uuid}/${locale}/confirmationFile`;
      default:
        return `${API_BASE_URL}/${TRANSACTIONS}`;
    }
  },
};
