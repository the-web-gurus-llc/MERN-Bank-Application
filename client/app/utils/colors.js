export default {
  white: '#fff',
  black: '#000',
  primaryBlue: '#1890ff',
  grey: 'rgb(242, 244, 247)',
  red: 'rgb(229, 0, 0)',
  redErr: '#ff4d4f',
  silver: '#bababa',
};
