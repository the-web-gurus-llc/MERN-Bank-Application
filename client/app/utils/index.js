export { default as colors } from './colors';
export { default as media } from './media';
export { default as request } from './request';
export { default as typography } from './typography';
export { default as api } from './api';
export { default as routes } from './routes';
export { default as locales } from './locales';
export { default as i18n } from './i18n';
