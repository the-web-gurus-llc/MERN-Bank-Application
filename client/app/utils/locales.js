exports.DEFAULT_LOCALE = 'en';
exports.DEFAULT_DATE_FORMAT = 'dd.MM.yyyy, hh:mm aa';

// prettier-ignore
exports.appLocales = [
  'en',
  'de',
  'pl',
];
