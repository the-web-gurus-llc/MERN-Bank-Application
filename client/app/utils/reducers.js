/**
 * Combine all reducers in this file and export the combined reducers.
 */

import { combineReducers } from 'redux';
import { connectRouter } from 'connected-react-router';

import history from 'utils/history';
import globalReducer from 'containers/App/reducer';
import languageProviderReducer from 'providers/LanguageProvider/reducer';
import loadingProviderReducer from 'providers/LoadingProvider/reducer';
import errorProviderReducer from 'providers/ErrorProvider/reducer';

/**
 * Merges the main reducer with the router state and dynamically injected reducers
 */
export default function createReducer(injectedReducers = {}) {
  const rootReducer = combineReducers({
    global: globalReducer,
    language: languageProviderReducer,
    loading: loadingProviderReducer,
    error: errorProviderReducer,
    router: connectRouter(history),
    ...injectedReducers,
  });

  return rootReducer;
}
