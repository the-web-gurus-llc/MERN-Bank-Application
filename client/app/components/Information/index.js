/**
 *
 * Information
 *
 */

import React from 'react';
import { FormattedMessage } from 'react-intl';
import messages from './messages';
import StyledInformation from './styles';

export default function Information() {
  return (
    <StyledInformation>
      <FormattedMessage {...messages.content} />
    </StyledInformation>
  );
}
