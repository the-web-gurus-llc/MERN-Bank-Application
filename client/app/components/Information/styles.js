import styled from 'styled-components';
import { media, colors } from 'utils';

const StyledInformation = styled.div`
  max-width: 1260px;
  margin: 10px auto;
  color: ${colors.white};
  background-color: ${colors.primaryBlue};
  padding: 10px 15px;
  border-radius: 2px;

  ${media.tablet`
    padding: 10px 130px;
  `}
`;

export default StyledInformation;
