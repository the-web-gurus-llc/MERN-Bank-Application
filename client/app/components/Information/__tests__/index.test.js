import React from 'react';
import { render } from '@testing-library/react';
import { createMemoryHistory } from 'history';
import { Provider } from 'react-redux';
import { IntlProvider } from 'react-intl';
import { DEFAULT_LOCALE } from 'utils/locales';
import configureStore from 'utils/configureStore';
import Information from '../index';

describe('<Information />', () => {
  const history = createMemoryHistory();
  const store = configureStore({}, history);

  it('should render a information', () => {
    const { container } = render(
      <Provider store={store}>
        <IntlProvider locale={DEFAULT_LOCALE}>
          <Information />
        </IntlProvider>
      </Provider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
