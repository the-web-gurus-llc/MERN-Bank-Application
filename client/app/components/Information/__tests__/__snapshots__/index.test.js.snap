// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Information /> should render a information 1`] = `
<div
  class="Informationstyle__StyledInformation-sc-1p8zm62-0 dDYeav"
>
  <span>
    If you take advantage of our promotion and register your account by the end of this year, you will receive 10,00 USD from us. Accounts already created will also receive this from us.
  </span>
</div>
`;
